src/test/java/io/pinecone/ListEndpointValidationTest.java

```diff
@@ -0,0 +1,104 @@
+package io.pinecone;
+
+import io.pinecone.clients.Index;
+import io.pinecone.configs.PineconeConnection;
+import io.pinecone.exceptions.PineconeValidationException;
+import io.pinecone.proto.VectorServiceGrpc;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ListEndpointValidationTest {
+
+    private String indexName;
+    private PineconeConnection connectionMock;
+    private VectorServiceGrpc.VectorServiceBlockingStub stubMock;
+    private Index index;
+
+    @BeforeAll
+    public void setUp() {
+        indexName = "test-index";
+
+        // Mock sync Pinecone connection
+        connectionMock = mock(PineconeConnection.class);
+        stubMock = mock(VectorServiceGrpc.VectorServiceBlockingStub.class);
+        when(connectionMock.getBlockingStub()).thenReturn(stubMock);
+
+        index = new Index(connectionMock, indexName);
+    }
+
+    // TODO: write one for no args
+    // TODO: write ones for async
```

Comment:
Stray TODOs 👀

Reply:
yes yes sry this commit was just to show you a preview of the no-args version, so it wasn't formal. Will clean up!